runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -130,6 +130,8 @@
 		"You feel something weird sneaking into your backpack",
 		"You have a funny feeling like you would have been followed");
 
+	private String screenShotSubDir = "Manual Screenshots";
```

Comment:
Instead of maintaining a field for the screenshot subdirectory, add a new signature to `takeScreenshot` which accepts a filename, subdirectory, and image. That way, these extra `screenShotSubDir = "Manual Screenshots";` resets can be removed and screenshot function calls can be more declarative.

Reply:
I still need a variable to track the the sub directory, are you suggesting just make it local to the function in which the subdirectory is created? I couldn't just pass the subdirectory name (for reasons like in the code below) where the takeScreenshot is called after a switch statement so I would need store the subdirectory based on the switch case.

https://github.com/runelite/runelite/blob/150d8722a889df943a5e3f7bc9811e863dc60d87/runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java#L517-L555

Reply:
I agree, a variable will be needed for the `GameTick` and `WidgetLoaded` subscribers, however this doesn't need to be a class-wide field.

Reply:
Ok I will see what I can do.